### DISCUSSION TITLE
Dependency Error while starting

### DISCUSSION THREAD

**Discussion by fac3m4n** [rubiin/ultimate-nest#420] [Q&A] [answered]
Hello, I was following instruction, removed not used modules. While starting following dependency error appears.

```shell
[Nest] 60283  - 04/28/2024, 2:59:41 PM   ERROR [ExceptionHandler] Nest can't resolve dependencies of the UserService (?, ReferralRepository, SqlEntityManager, ConfigService, AmqpConnection, CloudinaryService, MailerService). Please make sure that the argument "UserRepository" at index [0] is available in the UserModule context.

Potential solutions:
- Is UserModule a valid NestJS module?
- If "UserRepository" is a provider, is it part of the current UserModule?
- If "UserRepository" is exported from a separate @Module, is that module imported within UserModule?
  @Module({
    imports: [ /* the Module containing "UserRepository" */ ]
  })
``` 
Can you help on it?

**Comment by rubiin** [comment]
can you screenshot the constructor for userservice @fac3m4n

**Comment by mndev222** [comment]
I have the same problem.

> can you screenshot the constructor for userservice @fac3m4n

![image](https://github.com/user-attachments/assets/daf21517-91f4-4e99-a796-55dc029a04c5)
I can't run a dev server:
https://github.com/rubiin/ultimate-nest/discussions/482

**Reply by rubiin** [reply]
seems like an issue with mikroorm upgrade, will fix it asap

**Reply by rubiin** [reply]
related https://github.com/mikro-orm/nestjs/discussions/184 mostly affects the sql drivers imo

**Reply by fac3m4n** [reply]
was this fixed?

**Reply by mndev222** [reply]
I don't know how to fix it.

**Reply by rubiin** [reply]
@fac3m4n are you still facing this. I did a fresh install and didnt encounter it

**Reply by fac3m4n** [reply, marked as answer]
all good after fresh install 💪

**Comment by fac3m4n** [comment]
Make a fresh install

**Reply by rubiin** [reply]
Will be pushing a small update for deps and maybe rewritting the whole readme in a couple of days

**Reply by mndev222** [reply]
BTW. deploy.sh is not passing ENV to a docker-compose file when running a docker dev server.
